contrib/renew-client-cert

```diff
@@ -0,0 +1,66 @@
+#!/bin/sh -e
+
+# This is an OnChanged handler that can handle requests to renew the client
+# certificates used by the device to connect to the device gateway
+
+[ -z "$CONFIG_FILE" ] && (echo "No CONFIG_FILE specified"; exit 1)
+[ -f "$CONFIG_FILE" ] || (echo "$CONFIG_FILE does not exist"; exit 1)
+[ -z "$SOTA_DIR" ] && (echo "No SOTA_DIR specified"; exit 1)
+[ -d "$SOTA_DIR" ] || (echo "$SOTA_DIR does not exist"; exit 1)
+
+fioconfig=$(which fioconfig)
+
+timestamp=$(cat $CONFIG_FILE | grep TIMESTAMP | cut -d= -f2)
+[ -z "$timestamp" ] && (echo "No TIMESTAMP found in config file"; exit 1)
+echo "Timestamp for rotation is $timestamp"
+
+est_server=$(cat $CONFIG_FILE | grep ESTSERVER | cut -d= -f2)
+[ -z "$est_server" ] && (echo "No ESTSERVER found in config file"; exit 1)
+echo "EST server for rotation is $est_server"
+
+# We need to toggle between two different pkcs11 slot ids to rotate on HSMs:
+pkey_ids=$(cat $CONFIG_FILE | grep PKEYIDS | cut -d= -f2)
+if [ -n "$pkey_ids" ]; then
+	echo "PKCS11 key IDs: ${pkey_ids}"
+fi
+cert_ids=$(cat $CONFIG_FILE | grep CERTIDS | cut -d= -f2)
+if [ -n "$cert_ids" ]; then
+	echo "PKCS11 cert IDs: ${cert_ids}"
+fi
+
+rotate() {
+	extra_args=""
+	if [ -n "$pkey_ids" ]; then
+		extra_args="--pkcs11-key-ids=$pkey_ids"
+	fi
+	if [ -n "$cert_ids" ]; then
+		extra_args="$extra_args --pkcs11-cert-ids=$cert_ids"
+	fi
+	exec $fioconfig -c $SOTA_DIR renew-cert $est_server "$extra_args"
+}
+
+# We'll execute under a few conditions:
+# 1) A cert rotation has never taken place (cert-rotation.state.completed does not exist)
+# 2) We need to do a new rotation ($timestamp > timestamp of cert-rotation.state.completed)
+# 3) A rotation seems stuck (cert-rotation.state exists)
+
+active_file="${SOTA_DIR}/cert-rotation.state"
+completed_file="${active_file}.completed"
+
+if [ -f "$active_file" ] ; then
+	echo "Rotating certs: cert-rotation.state exists"
```

Comment:
If the cert rotation fails for some reason and then a user issues a new cert rotation request, then iiuc, the `rotate` at line #52 is executed which will continue the previous rotation with the previous context? It looks like it may lead to issues.

Reply:
fioconfig checks for this file at startup and will resume the rotation. So I think this isn't needed. Is this what you are alluding to?

Reply:
It will resume the previous rotation while a user has already issued the new cert rotation request. So, this handler starts with new config file and new param  so the cert rotation should start from scratch while according to this code it will be resumed .

Reply:
under the revised fioconfig, explicitly running `renew-certs` will start a new rotation even if the file exists. That old way was problematic for you, me and this code. 

But I still think i should drop this check. There's no value since fioconfig will have already tried to handle this.

Reply:
I think it should work.

I think the grepping and the IDs comparison can be dropped to, if you meant it in "But I still think i should drop this check."

Moreover, we may even drop the first `if`

Reply:
that code is mostly to help give more context to the logs when we hit support issues and also to save the remaining script from having to do ` || true` type stuff when it looks for content in the file